webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -357,14 +356,27 @@ final case class BlockStatusReport(
       // simple weighted compliance, as usual
       case WeightedReport => ComplianceLevel.sum(subComponents.map(_.compliance))
       // worst case bubble up, and its weight can be either 1 or the sum of sub-component weight
-      case worst: WorstReportReportingLogic =>
+      case worst: WorstReportWeightedReportingLogic =>
         val worstReport = ReportType.getWorseType(subComponents.map(_.status))
         val allReports  = getValues(_ => true).flatMap(_.messages.map(_ => worstReport))
         val kept        = worst match {
           case WorstReportWeightedOne => allReports.take(1)
           case WorstReportWeightedSum => allReports
         }
         ComplianceLevel.compute(kept)
+      case WorstReportByPercent =>
+        // Get reports of sub-components to find the worst by percent
+        val allReports = subComponents.map {
+          case b: BlockStatusReport =>
+            // Convert block compliance to percent, take the worst
+            b.getValues(_ => true).flatMap(_.messages.map(m => m.reportType))
+          case o: ValueStatusReport =>
+            // Value has 1 count
+            List(ReportType.getWorseType(o.getValues(_ => true).flatMap(_.messages.map(_ => o.status))))
```

Comment:
I'm not sure about the semantic of that one. Shouldn't we keep the weitght ? (ie is it more like weighted or like sum1 worst?)
It matters only for upper component, not directly that one.